syn/python/get_kge.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python3
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Parse a yosys area report and give a kGE equivalient
+
+import re
+import argparse
+
+weighted_dict = {}
+
+
+def read_lib(lib_file_path):
+    with open(lib_file_path, 'r') as f:
+        lib_file = f.readlines()
+
+    cell_dict = {}
+    cell_name = ''
+    for line in lib_file:
+        if re.search(r'^  cell \(', line):
+            if cell_name != '':
+                print("Size missing")
+                quit()
+            cell_name = line.split()[1].strip('()')
+        elif re.search(r'^\s+area', line):
+            if cell_name == '':
+                print("Size found without cell")
+                quit()
+            cell_area = line.split()[2].strip(';')
+            cell_dict[cell_name] = float(cell_area)
+            cell_name = ''
+    for cell in cell_dict:
+        weighted_dict[cell] = cell_dict[cell] / cell_dict['NAND2_X1']
+
+
+def get_kge(report_path):
+    with open(report_path, 'r') as f:
+        report = f.readlines()
+    ge = 0.0
+    for line in report:
+        data = line.split()
+        if len(data) == 0:
+            continue
+        if data[0] in weighted_dict:
+            ge += float(data[1]) * weighted_dict[data[0]]
```

Comment:
This is fine, but you could avoid the double lookup with:
```
weight = weighted_dict.get(data[0])
if weight is not None:
    ge += float(data[1]) * weight
```

And, as earlier, do you want to do any error checking to make sure there are exactly 2 words in the line and the second one is a valid float?

Reply:
Will do thanks